docs/optimization_guide.md

```diff
@@ -0,0 +1,66 @@
+alternative, what we call a 'pseudo-gradient'. The issue of updating can be resolved either by introducing latent weights, or by opting for a custom BNN optimizer.
+
+## Latent Weights
+
+Suppose we take a batch of training samples and evaluate a forward and backward pass. During the backward pass we replace the gradients with a pseudo-gradient, and we get a gradient vector on our weights. We then feed this into an optimizer like Adam, and get a vector with updates for our weights.
+
+At this point, what do we do? If we directly apply the updates to our weights, they are no longer binary. The standard solution to this problem has been to introduce real-valued **latent weights**. We apply our update step to this real-valued weight. During the forward pass, we use the binarized version of the latent weight.
+
+Beware that latent weights are [not really weights at all](https://arxiv.org/abs/1906.02107) - instead, they are best thought of as a product between the weight and a positive inertia: the higher this inertia, the stronger the signal required to make the weight flip.
+
+One implication of this is that the latent weights should be constrained: as an increase in inertia does not change the behavior of the network, it can otherwise grow indefinitely.
+
+In Larq, it is trivial to implement this strategy. An example of a layer optimized with this method would look like:
+
+```python
+x_out = larq.layers.QuantDense(512,
+                               input_qunatizer="ste_sign",
+                               kernel_quantizer="ste_sign",
+                               kernel_constraint="weight_clip")(x_out)
+```
+
+Any optimizer you now apply will update the latent weights; after the update the latent weights are clipped to $[-1, 1]$.
+
+### Alternative: Custom Optimizers
+
+Instead of using latent weights, one can opt for a custom BNN optimizer that inherently generates binary weights. An example of such an optimizer is [Bop](api/optimizers/#bop).
+
+## Choice of Pseudo-Gradient
+
+In [`larq.quantizers`](/api/quantizers) you will find a variety of quantizers that have been introduced in different papers. Many of these quantizers behave identically during the forward pass but implement different pseudo-gradients. Studies comparing different pseudo-gradients report little difference between them. Therefore, we recommend using the classical [`ste_sign()`](/api/quantizers/#ste_sign) as default.
+
+## Choice of Optimizer
+
+When using a latent weight strategy, you can apply any optimizer you are familiar with from real-valued deep learning. However, due the different nature of BNNs your intuitions may be off. We recommend using Adam: although other optimizers can achieve similar accuracies with a lot of finetuning, we and others have found that Adam is easiest to use for most types of networks.
+
```

Comment:
I would say that Adam is quickest to converge and least sensitive to the choice of hyperparamters